src/image/ImageManagerActions.cc

```diff
@@ -488,24 +498,52 @@ int ImageManager::delete_image(int iid, string& error_str)
         return -1;
     }
 
+    // Check associted VM state for backups. Note VM can no longer exist
+    // Fai if VM is in BACKUP and INCREMENT mode or FULL with no CONCURRENT_FORGET
+    if ( img->get_type() == Image::BACKUP )
+    {
+        auto ids   = img->get_running_ids();
+        auto first = ids.cbegin();
+
+        if (first != ids.cend())
+        {
+            VirtualMachinePool* vmpool = Nebula::instance().get_vmpool();
+
+            if (auto vm = vmpool->get_ro(*first))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that you are using unintended operations or assignments that could lead to logical errors, vulnerabilities, or undefined behavior. To prevent these issues, ensure that you follow best practices, such as using explicit comparisons and clear operator usage, and using parentheses to clarify intent. Learn more - https://cvefeed.io/cwe/detail/cwe-480-use-of-incorrect-operator